deployment/build-and-stage.yaml

```diff
@@ -17,13 +17,118 @@
 # eventually, they can probably be combined
 
 steps:
-# TODO(michaelkedar): Docker image creation & tagging
+- name: gcr.io/cloud-builders/git
+  args: ['submodule', 'update', '--init', '--recursive']
+  id: 'setup'
+
+# Poll currently-running Cloud Builds until this is the earliest-started build from this trigger/tag.
+# This is to avoid potential misordering of docker images and staging deployments
+# in the case when multiple commits to master occur close to one another.
+- name: gcr.io/cloud-builders/gcloud
+  entrypoint: bash
```

Comment:
Suggest making embedded shell snippets conform to https://google.github.io/styleguide/shellguide.html

The standout think I can see here in particular is https://google.github.io/styleguide/shellguide.html#s6.3-tests

Reply:
I've made it follow the style guide a bit more closely now.